internal/database/mysql_test.go

```diff
@@ -236,68 +263,98 @@ func TestDeleteUser(t *testing.T) {
+        {"GetNonExistentUser", 999, false, "", "", ""},
+    }
+
+    for _, tc := range testCases {
+        t.Run(tc.name, func(t *testing.T) {
+            user := subject.GetUser(tc.userID)
+            if tc.expectUser {
+                if user == nil {
+                    t.Fatal("Expected to find user, but got nil")
+                }
+                if user.Username != tc.username {
+                    t.Errorf("Expected username to be '%s', but got %s", tc.username, user.Username)
+                }
+                if user.Email != tc.email {
+                    t.Errorf("Expected email to be '%s', but got %s", tc.email, user.Email)
+                }
+                if err := bcrypt.CompareHashAndPassword([]byte(user.Password), []byte(tc.password)); err != nil {
+                    t.Errorf("Password does not match: %v", err)
+                }
+                if user.CreatedOn.IsZero() {
+                    t.Errorf("Expected created_on to be set, but got zero value")
+                }
+                if !user.IsValid() {
+                    t.Errorf("Expected valid user, but got invalid user: %v", user)
+                }
+            } else {
+                if user != nil {
+                    t.Errorf("Expected no user, but got: %v", user)
+                }
+            }
+        })
+    }
+}
 
-			// Get user ID by username
-			id := subject.GetUserID("username", "user2_id_user2")
-			if id != 2 {
-				t.Error("Failed to get user ID by username")
-			}
 
-			// Get user ID by email
-			id = subject.GetUserID("email", "user2_id_user2@example.com")
-			if id != 2 {
-				t.Error("Failed to get user ID by email")
-			}
-		},
-	)
-
-	t.Run(
-		"NonExistent_User", func(t *testing.T) {
-			// Get user ID by username
-			id := subject.GetUserID("username", "user3_id_user3")
-			if id != 0 {
-				t.Error("Expected to not find a user by username, but stored ID")
-			}
-
-			// Get user ID by email
-			id = subject.GetUserID("email", "user3_id_user3@example.com")
-			if id != 0 {
-				t.Error("Expected to not find a user by email, but stored ID")
-			}
-		},
-	)
+func TestGetUserID(t *testing.T) {
+    subject, probe := setup(t)
+    defer teardown(t, subject, probe)
+
+    t.Log("Adding user: user_id_user")
+    if err := subject.AddUser("user_id_user", "user_id_user@example.com", "user_id_pass"); err != nil {
+        t.Fatal("Failed to add user -", err)
+    }
+
+    t.Log("Adding user: user2_id_user2")
+    if err := subject.AddUser("user2_id_user2", "user2_id_user2@example.com", "user2_id2_pass"); err != nil {
+        t.Fatal("Failed to add user -", err)
+    }
+
+    testCases := []struct {
+        name     string
+        field    string
+        value    string
+        expected int
+    }{
+        {"GetUserIDByUsername1", "username", "user_id_user", 1},
+        {"GetUserIDByEmail1", "email", "user_id_user@example.com", 1},
+        {"GetUserIDByUsername2", "username", "user2_id_user2", 2},
```

Comment:
GetUser2IDByUsername, etc... Basically, the number should be labeling which user, not just apended at the end of the test name for clarity. Also, I thought we have ben using all_small_letter_and_underscores_for_test_names, probably best to keep one format

Reply:
All test descriptions are Camel Case, and test variables/entrie/data are in the all_small_letter_and_underscore format. Eg. test_pass for password is in that format.